Test/git-issues/git-issue-2068.dfy

```diff
@@ -0,0 +1,8 @@
+// RUN: %dafny_0 /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+type St = s:St_ | (forall o | o in s :: o.i(this))
```

Comment:
As per my other comment, it would be great to show this fix working correctly in a few other illegal contexts as well, just for deeper testing coverage.

Reply:
I added two other illegal context, the witness clause and a static const RHS. The static const RHS was also crashing Boogie. I was able to ensure the fixes use the same convention as for the existing code.
Of the top-level contexts, there is no other case to my knowledge where "this" should be disallowed.